README.md

```diff
@@ -1,118 +1,254 @@
 
-2. Indexing files on creation in object storage.
+### I want to associate Indexd data to structured data in a Gen3 Data Commons
 
-Using AWS SNS or Google PubSub it is possible to have streaming notifications when files are created, modified or deleted in the respective cloud object storage services (S3, GCS). It is then possible to use an AWS Lambda or GCP Cloud Function to automatically index the new object into Indexd. This may require using the batch processing services on AWS if the file is large to compute the necessary minimal set of hashes to support indexing. This feature can be set up on a per commons basis for any buckets of interest. The buckets do not have to be owned by the commons, but permissions to read the bucket objects and permissions for SNS or PubSub are necessary.
+> NOTE: This assumes that the data already exists in storage location(s)
+
+#### Indexd Record Creation Through Gen3's Data Submission Service: [Sheepdog](https://github.com/uc-cdis/sheepdog)
+
+When data files are submitted to a Gen3 Data Commons using Sheepdog, the files are automatically indexed into Indexd. Submissions to Sheepdog can include `object_id`'s that map to existing Indexd GUIDs. Or, if there are no existing records, Sheepdog can create them on the fly.
+
+To create Indexd records on the fly, Sheepdog will check if the file being submitted has a hash & file size matching anything currently in Indexd and if so uses the returned document GUID as the object ID reference. If no match is found in Indexd then a new record is created and stored in Indexd.
+
+### I want to index data that is dynamically added to storage location(s)
+
+#### Automatically Creating Indexd Records when Objects are Added to Object Storage
+
+Using AWS SNS or Google PubSub it is possible to have streaming notifications when files are created, modified or deleted in the respective cloud object storage services (S3, GCS). It is then possible to use an AWS Lambda or GCP Cloud Function to automatically index the new object into Indexd.
+
+> NOTE: This may require using the batch processing services if the file is large (to compute the necessary minimal set of hashes to support indexing). There are known limitations with AWS Lambda and GCP Cloud Functions related to how long a process can run before AWS/Google cuts it off. Some hash calculations may exceed that time limit.
+
+This feature can be set up on a per Data Commons basis for any buckets of interest. The buckets do not have to be owned by the commons, but permissions to read the bucket objects and permissions for SNS or PubSub are necessary.
 
 For existing data in buckets, the SNS or PubSub notifications may be simulated such that the indexing functions are started for each object in the bucket. This is useful because only a single code path is necessary for indexing the contents of an object.
 
-3. Indexing void object for fully control the bucket structure.
+## Indexd REST API for Record Creation
```

Comment:
this section could be a subsection of https://github.com/uc-cdis/indexd/tree/chore/docs#i-want-to-associate-indexd-data-to-structured-data-in-a-gen3-data-commons: either create records through sheepdog or directly in indexd

Reply:
ooh, good point

Reply:
well, it doesn't only fit under the use case there though. I think I wanna have this outside for more visibility